### PR TITLE
832 bug wrong schedule index

### LINKED ISSUE #832: API_'Schedules'_Status code 500 ('Internal Server Error' message) is displayed when User adds new schedule with wrong 'index' data
**Environment:** Windows 10
**Reproducible:** always
**Build found**: last commit from https://github.com/ita-social-projects/WhatBackend (dev branch)

**Pre-conditions:**

1. Log in as Administrator to https://whatbackend.azurewebsites.net/index.html?urls.primaryName=WHAT%20Project%20API%20v2

**Steps to reproduce:**

1. Make POST request to REST endpoint https://whatbackend.azurewebsites.net/api/v2/schedules with wrong 'index' data, other values are valid. Request body contains JSON object:

{
"pattern":
{ "type": 3, "interval": 1, "daysOfWeek": [ 3 ], "index": 6 },
"range":
{ "startDate": "2022-04-01T10:00:00.000Z", "finishDate": "2022-05-31T10:00:00.000Z" },
"context":
{ "groupID": 1, "themeID": 1, "mentorID": 1 }
}

**Actual result:**

Status code: 500 Undocumented. Error: Internal Server Error.

**Expected result:**

Status code: 400. Error: Can not create schedule due to wrong request data.

**Notes**

The same bug also is reproduced for the following conditions:

- non-existent mentorID, other values are valid.
- wrong 'type' data (4 and bigger), other values are valid.

Please see the attachments for these bugs.

**User story and test case links:**
Test case https://jira.softserve.academy/browse/DW-367
SRS https://github.com/ita-social-projects/WhatBackend/wiki/Schedule-service-overview

**Labels to be added:**
Priority ('Medium'), Severity ('Critical').

https://user-images.githubusercontent.com/97453653/161383528-5384a137-b7e3-45c3-a0dc-3d7d2a610ef1.mp4

https://user-images.githubusercontent.com/97453653/161383550-07318aa1-93d2-4e1f-ba07-65bb555e39a4.mp4

https://user-images.githubusercontent.com/97453653/161383555-8dc8c919-0ae3-4461-a9a5-8611b74a7fb4.mp4

### PR DESCRIPTION
**Pull request by BobTheSuperman**
Fixed "API_'Schedules'_Status code 500 ('Internal Server Error' message) is displayed when User adds new schedule with wrong 'index' data"

### CODE REVIEW

File: CharlieBackend.Business/Helpers/ResponseMessages.cs [hunk 1]
```diff
@@ -1,12 +1,15 @@
-﻿using System;
-using System.Collections.Generic;
-using System.Text;
-
-namespace CharlieBackend.Business.Helpers
+﻿namespace CharlieBackend.Business.Helpers
 {
     public class ResponseMessages
```
Comment (PavelTsyposhchuk) [inline CharlieBackend.Business/Helpers/ResponseMessages.cs]: make this class static, we dont need instance this class

File: CharlieBackend.Business/Services/ScheduleServiceFolder/Helpers/SchedulesEventsDbEntityVerifier.cs [hunk 2]
```diff
@@ -23,22 +25,34 @@ public async Task<string> ValidateCreateScheduleRequestAsync(CreateScheduleDto r
...
+                error.Append(ResponseMessages.NotExist("Group"));
             }
 
-            if (request.Context.MentorID.HasValue && !await _unitOfWork.MentorRepository.IsEntityExistAsync(request.Context.MentorID.Value))
+            if (request.Context.MentorID.HasValue && await _unitOfWork.MentorRepository.IsEntityExistAsync(request.Context.MentorID.Value))
             {
-                error.Append(" Mentor does not exist");
+                if ((await _unitOfWork.AccountRepository.GetAccountCredentialsById(request.Context.MentorID.Value)).IsActive == false)
+                {
+                    error.Append(" Mentor is not active");
```
Comment (PavelTsyposhchuk) [inline CharlieBackend.Business/Services/ScheduleServiceFolder/Helpers/SchedulesEventsDbEntityVerifier.cs]: U expand class ResponseMessages, but used there usually string, why don't create such message in class?

**Review by PavelTsyposhchuk — CHANGES_REQUESTED** [review]
see comments added, write in TG in case any questions

File: CharlieBackend.Business/Helpers/ResponseMessages.cs (hunk 1 above)
Comment (BobTheSuperman) [reply CharlieBackend.Business/Helpers/ResponseMessages.cs]: Done

File: CharlieBackend.Business/Services/ScheduleServiceFolder/Helpers/SchedulesEventsDbEntityVerifier.cs (hunk 2 above)
Comment (BobTheSuperman) [reply CharlieBackend.Business/Services/ScheduleServiceFolder/Helpers/SchedulesEventsDbEntityVerifier.cs]: The corresponding message is added to the ResponseMessages class.

File: CharlieBackend.Business/Helpers/ResponseMessages.cs [hunk 3]
```diff
@@ -1,12 +1,17 @@
...
+
         public const string GroupHasNotStudents = "Group hasn't any students";
+
+        public const string IndexNotValid = "Index cannot be less then 1 and greater than 5";
+
+        public static string NotExist(string entityName) => $"{entityName} does not exist ";
+
+        public static string NotValid(string entityName) => $"{entityName} is not valid ";
+
+        public static string NotActive(string entityName) => $"{entityName} is not active ";
```
Comment (PavelTsyposhchuk) [inline CharlieBackend.Business/Helpers/ResponseMessages.cs:15]: maybe better if u make all members looks the same(all with static keyword or all without)

**Review by PavelTsyposhchuk — APPROVED** [review]
only cosmetic comment on ur own decision to change according them or not

**Review by Ssashaa — APPROVED** [review]

**Review by da-real-ivanpan — APPROVED** [review]

**Review by filinskyi — APPROVED** [review]
LGTM

**Review by da-real-ivanpan — APPROVED** [review]

**Review by BohdanZaliubovskyi — APPROVED** [review]

**Pull request merged**